src/Codeception/Module/DataFactory.php

```diff
@@ -0,0 +1,169 @@
+<?php
+namespace Codeception\Module;
+
+use Codeception\Lib\Interfaces\DataMapper;
+use Codeception\Lib\Interfaces\ORM;
+use Codeception\TestCase;
+use League\FactoryMuffin\FactoryMuffin;
+use League\FactoryMuffin\Stores\RepositoryStore;
+
+/**
+ * DataFactory allows you to easily generate and create test data using [**FactoryMuffin**](https://github.com/thephpleague/factory-muffin).
+ * DataFactory uses an ORM of your application to define, save and cleanup data. Thus, should be used with ORM or Frqmework modules.
+ *
+ * Generation rules can be defined in a factories file. You will need to create `factories.php` (it is recommended to store it in `_support` dir)
+ * Follow [FactoryMuffin documentation](https://github.com/thephpleague/factory-muffin) to set valid rules.
+ * Random data is set with [Faker](https://github.com/fzaninotto/Faker) library.
+ *
+ * ```php
+ * use League\FactoryMuffin\Faker\Facade as Faker;
+ *
+ * // MyModel is a valid class name of a model you are using
+ * $fm->define('MyModel')->setDefinitions([
+ *
+ *  'name'   => Faker::name(),
+ *    'email'  => Faker::email(),
+ *    'body'   => Faker::text()
+ *]);
+ * ```
+ *
+ * Configure this module to load factories (you can set several factories files to use).
+ * You should also specify a module with an ORM as a dependency.
+ *
+ * ```yaml
+ * modules:
+ *     enabled:
+ *         - DataFactory:
+ *             factories: ['_support/factories.php']
+ *             depends: Laravel5
+ * ```
+ *
+ * (you can also use Yii2, Phalcon, or Doctrine)
+ *
+ * Alternatively you can define factories from a Helper class with `_define` method.
+ *
+ * ```php
+ * function _initialize()
+ * {
+ *      $this->getModule('DataFactory')->_define('MyModule', [
+ *          'name' => Faker::name()
+ *      ]);
+ * }
+ * ```
+ *
+ *
+ */
+class DataFactory extends \Codeception\Module
+{
+    protected $dependencyMessage = <<<EOF
+ORM module (like Doctrine2) or Framework module with ActiveRecord support is required:
+--
+modules:
+    enabled:
+        - DataFactory:
+            depends: Laravel5
+            factories: [_support/factories.php]
+--
+EOF;
+
+    protected $orm;
+
+    /**
+     * @var FactoryMuffin
+     */
+    public $factoryMuffin;
+
+    protected $config = ['factories' => null];
+
+    public function _initialize()
+    {
+        if (!class_exists('League\FactoryMuffin\FactoryMuffin')) {
+            throw new \Exception('FactoryMuffin not installed. Please add `"league/factory-muffin": "~3.0"` to composer.json');
+        }
+
+        $store = null;
+        if ($this->orm instanceof DataMapper) { // for Doctrine
+            $store = new RepositoryStore($this->orm->_getEntityManager());
```

Comment:
Ah, ok, ModelStore is used by default and default methods match Eloquent methods, so Laravel should be working.